src/main/java/net/ccbluex/liquidbounce/file/configs/FriendsConfig.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * LiquidBounce Hacked Client
+ * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
+ * https://github.com/CCBlueX/LiquidBounce/
+ */
+package net.ccbluex.liquidbounce.file.configs
+
+import com.google.gson.*
+import net.ccbluex.liquidbounce.file.FileConfig
+import net.ccbluex.liquidbounce.file.FileManager.PRETTY_GSON
+import net.ccbluex.liquidbounce.utils.ClientUtils.LOGGER
+import java.io.*
+
+class FriendsConfig(file: File) : FileConfig(file) {
+    val friends: MutableList<Friend> = ArrayList()
+
+    /**
+     * Load config from file
+     *
+     * @throws IOException
+     */
+    @Throws(IOException::class)
+    override fun loadConfig() {
+        clearFriends()
+        try {
+            val jsonElement = JsonParser().parse(file.bufferedReader())
+            if (jsonElement is JsonNull) return
+            for (friendElement in jsonElement.asJsonArray) {
+                val friendObject = friendElement.asJsonObject
+                addFriend(friendObject["playerName"].asString, friendObject["alias"].asString)
+            }
+        } catch (ex: JsonSyntaxException) {
+            //When the JSON Parse fail, the client try to load and update the old config
+            LOGGER.info("[FileManager] Try to load old Friends config...")
+            val bufferedReader = file.bufferedReader()
+            var line: String
+            while (bufferedReader.readLine().also { line = it } != null) {
+                if ("{" !in line && "}" !in line) {
+                    line = line.replace(" ", "").replace("\"", "").replace(",", "")
+                    if (":" in line) {
+                        val data = line.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+                        addFriend(data[0], data[1])
+                    } else addFriend(line)
+                }
+            }
+            bufferedReader.close()
+            LOGGER.info("[FileManager] Loaded old Friends config...")
+
+            //Save the friends into a new valid JSON file
+            saveConfig()
+            LOGGER.info("[FileManager] Saved Friends to new config...")
+        } catch (ex: IllegalStateException) {
+            LOGGER.info("[FileManager] Try to load old Friends config...")
+            val bufferedReader = file.bufferedReader()
+            var line: String
+            while (bufferedReader.readLine().also { line = it } != null) {
+                if ("{" !in line && "}" !in line) {
+                    line = line.replace(" ", "").replace("\"", "").replace(",", "")
+                    if (":" in line) {
+                        val data = line.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+                        addFriend(data[0], data[1])
+                    } else addFriend(line)
+                }
+            }
+            bufferedReader.close()
```

Comment:
This could have been done completly different using Kotlin. But might not be relevant anymore anyway